external/llvm-project/mlir/lib/Dialect/XModel/Transforms/AsyncGraph.cpp

```diff
@@ -73,8 +73,10 @@ class XModelAsyncGraphPass
     // Insert host sync before operation that reads an async::value
     for (auto operand : op->getOperands()) {
       if (auto itoken = res2tokens.lookup(operand)) {
-        createWaitOp(op, itoken);
-        currentTokens.erase(itoken);
+        if (currentTokens.contains(itoken)) {
```

Comment:
What's this change for?

Reply:
Because there seemed to be an assumption token --> output operand to be 1:1. 
w/o this it will end up creating two awaits on the same token.

Reply:
Ah